vendor/github.com/raintank/schema/metric.go

```diff
@@ -81,6 +85,23 @@ func (m *MetricData) KeyBySeries(b []byte) []byte {
 	return b
 }
 
+func (m *MetricData) KeyBySeriesWithTags(b []byte) []byte {
+	nameWithTagsBuffer := bytes.NewBuffer(b)
+	nameWithTagsBuffer.WriteString(m.Name)
+
+	sort.Strings(m.Tags)
+	for _, t := range m.Tags {
+		if len(t) >= 5 && t[:5] == "name=" {
+			continue
+		}
+
+		nameWithTagsBuffer.WriteString(";")
+		nameWithTagsBuffer.WriteString(t)
+	}
+
+	return nameWithTagsBuffer.Bytes()
```

Comment:
@robert-milan do you think we can make this better?

Reply:
right, we could use a pool but that's a concern for the caller.
I should have been clearer, i specifically wonder:
1) if you think it's silly to use ascii separators like `;` and not `0x00` or something . i don't see any downside to using something ascii friendly. Though i will say since it's only a single char we should probably use WriteRune instead of WriteString
2) is there anything there to take into account wrt the upcoming interning. To be precise, i don't want to be in a situation where the output is different once we support interning, or where keeping the output the same makes the implementation suboptimal

Reply:
@robert-milan ^

Reply:
I might be missing something, as I have not reviewed the PR in great detail, but as I understand it this is only used for partitioning. My PR deals strictly with interning in the index. The only time I touch any `MetricData` is when I convert it into a `MetricDefinition`, so I don't think the two affect each other.

1. I don't think it matters. Also, I would use `WriteByte` instead of `WriteRune`. Are we worried about needing an escape sequence (like when processing stream data) that we have to scan for or something? That would require a different answer.

2. Since this only appears to operate on `MetricData` I don't think it will be an issue.